lib/JpegTran.js

```diff
@@ -1,33 +1,40 @@
 var childProcess = require('child_process'),
     Stream = require('stream').Stream,
     util = require('util'),
-    which = require('which'),
-    memoizeAsync = require('memoizeasync');
+    which = require('which');
 
-function JpegTran(jpegTranArgs) {
+function JpegTran(jpegTranArgs, jpegTranBinary) {
     Stream.call(this);
 
     this.jpegTranArgs = jpegTranArgs;
 
     this.writable = this.readable = true;
 
     this.hasEnded = false;
+
+    this.jpegBinary = jpegTranBinary;
 }
 
 util.inherits(JpegTran, Stream);
 
-JpegTran.getBinaryPath = memoizeAsync(function (cb) {
+JpegTran.prototype.getBinaryPath = function (defaultBinary, cb) {
+    if (defaultBinary) {
+        cb(null, defaultBinary);
```

Comment:
You're [releasing Zalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony), please call the callback in a `setImmediate`.